packages/ember-glimmer/tests/integration/helpers/get-test.js

```diff
@@ -0,0 +1,337 @@
+    this.assertText('[yellow] [yellow]');
+
+    this.runTask(() => set(this.context, 'colors', { apple: 'green', banana: 'purple' }));
+    this.assertText('[purple] [purple]');
+
+    this.runTask(() => set(this.context, 'key', 'key1'));
+    this.assertText('[green] [green]');
+
+    this.runTask(() => set(this.context, 'colors.apple', 'red'));
+    this.assertText('[red] [red]');
+  }
+
+  ['@test should be able to get an object value with a GetStream value and bound/dynamic key']() {
+    this.render(`[{{get (get possibleValues objectKey) key}}] [{{if true (get (get possibleValues objectKey) key)}}]`, {
+      possibleValues: {
+        colors1: { apple: 'red', banana: 'yellow' },
+        colors2: { apple: 'green', banana: 'purple' }
+      },
+      objectKey: 'colors1',
+      key: 'apple'
+    });
+    this.assertText('[red] [red]');
+
+    this.runTask(() => set(this.context, 'objectKey', 'colors2'));
+    this.assertText('[green] [green]');
+
+    this.runTask(() => set(this.context, 'objectKey', 'colors1'));
+    this.assertText('[red] [red]');
+
+    this.runTask(() => set(this.context, 'key', 'banana'));
+    this.assertText('[yellow] [yellow]');
+
+    this.runTask(() => set(this.context, 'objectKey', 'colors2'));
+    this.assertText('[purple] [purple]');
+
+    this.runTask(() => set(this.context, 'objectKey', 'colors1'));
+    this.assertText('[yellow] [yellow]');
+  }
+
+  ['@test should be able to get an object value with a GetStream value and GetStream key']() {
+    this.render(`[{{get (get possibleValues objectKey) (get possibleKeys key)}}] [{{if true (get (get possibleValues objectKey) (get possibleKeys key))}}]`, {
+      possibleValues: {
+        colors1: { apple: 'red', banana: 'yellow' },
+        colors2: { apple: 'green', banana: 'purple' }
+      },
+      objectKey: 'colors1',
+      possibleKeys: {
+        key1: 'apple',
+        key2: 'banana'
+      },
+      key: 'key1'
+    });
+    this.assertText('[red] [red]');
+
+    this.runTask(() => set(this.context, 'objectKey', 'colors2'));
+    this.assertText('[green] [green]');
+
+    this.runTask(() => set(this.context, 'objectKey', 'colors1'));
+    this.assertText('[red] [red]');
+
+    this.runTask(() => set(this.context, 'key', 'key2'));
+    this.assertText('[yellow] [yellow]');
+
+    this.runTask(() => set(this.context, 'objectKey', 'colors2'));
+    this.assertText('[purple] [purple]');
+
+    this.runTask(() => set(this.context, 'objectKey', 'colors1'));
+    this.assertText('[yellow] [yellow]');
+  }
+
+  ['@test should yield `get`']() {
+    let fooBarInstance;
+    let FooBarComponent = Component.extend({
+      init() {
+        this._super();
+        fooBarInstance = this;
+        this.mcintosh = 'red';
+      }
+    });
+
+    this.registerComponent('foo-bar', {
+      ComponentClass: FooBarComponent,
+      template: `{{yield (get colors mcintosh)}}`
+    });
+
+    this.render(`{{#foo-bar colors=colors as |value|}}{{value}}{{/foo-bar}}`, {
```

Comment:
To test the `get()` method on the reference, you would need to do `{{value.foo}}`

Actually, if that's the goal, there is a simpler test that doesn't require a component:

```
{{#with (get foo bar) as |baz|}}
  {{baz.bat}}
{{/with}}
```

The baz in `{{baz.bat}}` would be a `GetReference`, and `{{baz.bat}}` would call `bazGetReference.get('bat')`.